flagset/flagset.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2021 Google Inc. All Rights Reserved.
```

Comment:
I like pulling this interface out to avoid duplication between driver/driver.go and internal/plugin/plugin.go. Can we make this a separate PR? I think this can go in regardless of other discussions. One thing I'm not sure about is whether it's worth a separate top-level directory - seems too shallow for that. Maybe have it as driver/flagset.go? (and "package driver").

Reply:
> Can we make this a separate PR?

Yes.  We just need to decide on the package for flagset.go.

> Maybe have it as driver/flagset.go?

I had tried adding to driver package before, but ran into an import cycle.
package github.com/google/pprof
	imports github.com/google/pprof/driver
	imports github.com/google/pprof/internal/driver
	imports github.com/google/pprof/internal/binutils
	imports github.com/google/pprof/internal/plugin
	imports github.com/google/pprof/driver: import cycle not allowed
package github.com/google/pprof
	imports github.com/google/pprof/driver
	imports github.com/google/pprof/internal/driver
	imports github.com/google/pprof/internal/binutils
	imports github.com/google/pprof/internal/plugin
	imports github.com/google/pprof/driver: import cycle not allowed

Are you OK going with a new package?  I could not think of any other reasonable place for this.

Reply:
I'd prefer not to have a new top-level package. Actually, reading go/go-style/decisions#interfaces I wonder if we should leave things as is with two copies of the interface for now. It's duplication, but it's rather unrelated issue to this PR, so I'd tackle it separately if we ever need to.